### DISCUSSION TITLE
How do fastest way load huge data

### DISCUSSION THREAD

**Discussion by MohMehrnia** [ravendb/ravendb#13121] [General]
How do the fastest way to load huge data from the server?
I need to load all records between two DateTime, is the fastest way to load data?
I use the below code, but when the date range is large loading very very slow

```
public async static Task<List<WellRecord>> LogSnapShot(string startDate, string endDate)
        {
            using (var store = new DocumentStore
            {
                Urls = new[] { ConnectionsSetting.RavenDBServer },
                Database = ConnectionsSetting.RavenDBLogDatabase,
                Certificate = cert,
                Conventions =
                {
                    UseCompression=true,
                }
            })
            {
                store.Initialize();
                using (IAsyncDocumentSession Session = store.OpenAsyncSession(new SessionOptions
                {
                    NoCaching = true,
                    NoTracking = true,
                    TransactionMode = TransactionMode.SingleNode
                }))
                {
                    var result = await Session.Query<WellRecord>().Where(q => q.Rdtime >= DateTime.Parse(startDate) && q.Rdtime <= DateTime.Parse(endDate)).ToListAsync();
                    return result;
                };
            }
        }
```

**Comment by haludi** [comment]
Hi,

How big are the result of the large range?
How big are the documents?
If you expect to get huge results you should use paging or stream queries.
https://ravendb.net/docs/article-page/5.2/csharp/indexes/querying/paging
https://ravendb.net/docs/article-page/5.2/csharp/client-api/session/querying/how-to-stream-query-results

(Also you can move the parsing outside of the query so it will be done once
but I am not sure how much it will add to the performance)

Best regards,
Igal

**Reply by ayende** [reply]
In addition, your code always creates the document store inline. that is a *bad* idea. You are expected to have a single document store per application, as a singleton instance.
